cluster-operator/src/test/java/io/strimzi/operator/cluster/model/CruiseControlTest.java

```diff
@@ -0,0 +1,578 @@
+import static io.strimzi.operator.cluster.model.cruisecontrol.Capacity.DEFAULT_BROKER_NW_OUT_CAPACITY;
+import static java.util.Collections.singletonMap;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CruiseControlTest {
+    private final String namespace = "test";
+    private final String cluster = "foo";
+    private final int replicas = 1;
+    private final String image = "my-image:latest";
+    private final int healthDelay = 120;
+    private final int healthTimeout = 30;
+    private final String minInsyncReplicas = "2";
+    private final Map<String, Object> metricsCm = singletonMap("animal", "wombat");
+    private final Map<String, Object> kafkaConfig = singletonMap(CruiseControl.MIN_INSYNC_REPLICAS, minInsyncReplicas);
+    private final Map<String, Object> zooConfig = singletonMap("foo", "bar");
+
+    CruiseControlConfiguration configuration = new CruiseControlConfiguration(new HashMap<String, Object>() {{
+            putAll(configuration.getCruiseControlDefaultPropertiesMap());
+            put("num.partition.metrics.windows", "2");
+        }}.entrySet()
+    );
+    private final Storage kafkaStorage = new EphemeralStorage();
+    private final SingleVolumeStorage zkStorage = new EphemeralStorage();
+    private final InlineLogging kafkaLogJson = new InlineLogging();
+    private final InlineLogging zooLogJson = new InlineLogging();
+    private final String version = KafkaVersionTestUtils.DEFAULT_KAFKA_VERSION;
+    private static final KafkaVersion.Lookup VERSIONS = KafkaVersionTestUtils.getKafkaVersionLookup();
+    private final String kafkaHeapOpts = "-Xms" + AbstractModel.DEFAULT_JVM_XMS;
+    private final String ccImage = "my-cruise-control-image";
+
+    {
+        kafkaLogJson.setLoggers(singletonMap("kafka.root.logger.level", "OFF"));
+        zooLogJson.setLoggers(singletonMap("zookeeper.root.logger", "OFF"));
+    }
+
+    private final CruiseControlSpec cruiseControlSpec = new CruiseControlSpecBuilder()
+            .withImage(ccImage)
+            .withReplicas(replicas)
+            .withConfig((Map) configuration.asOrderedProperties().asMap())
+            .build();
+
+    private final Kafka resource =
+            new KafkaBuilder(ResourceUtils.createKafkaCluster(namespace, cluster, replicas, image, healthDelay, healthTimeout))
+            .editSpec()
+                .editKafka()
+                    .withVersion(version)
+                    .withConfig(kafkaConfig)
+                .endKafka()
+                .withCruiseControl(cruiseControlSpec)
+            .endSpec()
+            .build();
+
+    private final CruiseControl cc = CruiseControl.fromCrd(resource, VERSIONS);
+
+    public void checkOwnerReference(OwnerReference ownerRef, HasMetadata resource)  {
```

Comment:
Just as an aside, this might be a good function to contribute to test utils as I can see all of the tests needing to call this method!

Reply:
I'll save this for another PR if that is alright with you two, the PR already changes a lot of files!

Reply:
@kyguy for sure updating ALL the other files should be done in a different PR but here you could start to move the method in the test utils and using it. Then, again, reusing the method for ALL the other tests has to be done in a different PR. Wdyt?